src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newStates = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': Object.assign(copyState, action.extraData);
+        newStates.push({ ...copyState });
```

Comment:
you can use ` push` only once in this task

Reply:
yeap, you`re right!